### PR TITLE
feat(fiori-elements-writer): removed `sap.fe.templates` for V4 FPM

### PR DESCRIPTION
**Pull request by martinjanus-sap**
#1942 

- FPM UI5 dependencies depend on `sap.fe.core` and not `sap.fe.templates` in the manifest.json
- Updated to load `sap.fe.templates` in local-ui5.yaml if it's V4 FPM app and doesn't already contain `sap.fe.templates` , as its needed to load local preview. 
- FPM to also load `sap.fe.templates` for flpSandbox.html
- Removed dependency on `sap.ushell` for V4 manifest.json.
   - `sap.ushell` loaded in local-ui5.yaml, needed to load local preview

### CODE REVIEW

**Comment by hitesh-parmar** [comment]
I'm not sure if I understand the change below. If the library is not needed in `manifest.json,` then it should also not be required in the YAML file. 

> Updated to load sap.fe.templates in local-ui5.yaml if it's V4 FPM app and doesn't already contain sap.fe.templates , as its needed to load local preview.

**Comment by martinjanus-sap** [comment]
> I'm not sure if I understand the change below. If the library is not needed in `manifest.json,` then it should also not be required in the YAML file.
> 
> > Updated to load sap.fe.templates in local-ui5.yaml if it's V4 FPM app and doesn't already contain sap.fe.templates , as its needed to load local preview.

 Both `sap.ushell` and `sap.fe.templates` is needed to run the local preview. So there is a difference there between manifest and ui5-local.yaml.
 
Comments on previous PR: 
- https://github.com/SAP/open-ux-tools/pull/1948#discussion_r1609940624
- https://github.com/SAP/open-ux-tools/pull/1948#discussion_r1613238304
 
 FYI @devinea

File: packages/fiori-elements-writer/src/fpmConfig.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+/**
+ * Processes the template for the Flexible Programming Model (FPM).
+ * Generates a custom page and updates the UI5 local yaml.
+ *
+ * @param feApp - the FE app config
+ * @param basePath - the absolute target path where the application will be generated
+ * @param fs - reference to a mem-fs editor
+ * @returns {Promise<void>}
+ */
+export async function fpmConfig<T extends {}>(feApp: FioriElementsApp<T>, basePath: string, fs: Editor): Promise<void> {
```
[the remark this answers is not in the document]
Comment (martinjanus-sap) [reply packages/fiori-elements-writer/src/fpmConfig.ts]: addressed in https://github.com/SAP/open-ux-tools/pull/2237/commits/c185db3fbb2ffdd8fca2aecb9e1ba7b9887a125e 
Thanks

File: packages/fiori-elements-writer/src/index.ts [hunk 2]
```diff
@@ -101,6 +101,10 @@ async function generate<T extends {}>(basePath: string, data: FioriElementsApp<T
     // Define template options with changes preview and loader settings based on project type
     const templateOptions = getTemplateOptions(isEdmxProjectType, feApp.service.version, feApp.ui5?.version);
 
+    // FPM to preload sap.fe.templates for FLP template
+    if (feApp.template.type === TemplateType.FlexibleProgrammingModel) {
+        (ui5Libs as string[]).push('sap.fe.templates');
```
Comment (IainSAP) [inline packages/fiori-elements-writer/src/index.ts:106]: Shouldnt ui5Libs be undefined for CAP (as per https://github.com/SAP/open-ux-tools/pull/2237/files#diff-a51a19d6fa0b94939e84e2a4eedd1b602cae40c2077c22ecce59ee537f5ccbcfR100) . If so I would suggest combining line: 100 and line 105-107 into a single function for clarity?

**Review by IainSAP — COMMENTED** [review]
One comment, otherwise LGTM. Thx @martinjanus-sap

File: packages/fiori-elements-writer/src/index.ts (hunk 2 above)
Comment (martinjanus-sap) [reply packages/fiori-elements-writer/src/index.ts]: addressed in https://github.com/SAP/open-ux-tools/pull/2237/commits/5b6e9f5ff7115d9558424cd2d49003b84500bb67
Thanks

**Review by IainSAP — APPROVED** [review]
Thanks @martinjanus-sap 

- Code is clean, abstraction level is appropriate for changes
- Tests updated to cover

**Review by devinea — APPROVED** [review]
Updates are clear and covered by tests. 
Changeset ✅

**Pull request merged**